examples/advanced/keycloak-site-authentication/site/local/custom/keycloak_security_handler.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple
+
+import json
+import jwt
+import requests
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_constant import FLContextKey
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.job_def import JobMetaKey
+
+
+class CustomSecurityHandler(FLComponent):
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.AUTHORIZE_COMMAND_CHECK:
+            result, reason = self.authorize(fl_ctx=fl_ctx)
+            if not result:
+                fl_ctx.set_prop(FLContextKey.AUTHORIZATION_RESULT, False, sticky=False)
+                fl_ctx.set_prop(FLContextKey.AUTHORIZATION_REASON, reason, sticky=False)
+
+    def _validate_token(self, token):
+        try:
+            response = requests.get("http://localhost:8080/realms/myrealm")
+            public_key = "-----BEGIN PUBLIC KEY-----\n" + \
+                         json.loads(response.text).get("public_key") + \
+                         "\n-----END PUBLIC KEY-----"
+
+            access_token_json = jwt.decode(token, public_key, options={"verify_signature": False})
```

Comment:
Why is verify_signature False?
@nvidianz , can you take a look at this validation logic? It seems that we get the public key from the IdP, and then use it to validate the token (which was issued by the same IdP).
I thought we should send the token to the IdP so that it would know who the user is, and any additional authz func could be done by the IdP.
Or since it's a JWT token, maybe the token already has some authz-related info, and we can validate it here?

Reply:
Public key should be configured here. It means we trust the IdP's token.

Reply:
Store the public key in "public_key.pem" file. And changed to "verify_signature" True.